src/jdk.jlink/share/classes/jdk/tools/jlink/internal/Platform.java

```diff
@@ -24,112 +24,58 @@
  */
 package jdk.tools.jlink.internal;
 
+import jdk.internal.util.Architecture;
+import jdk.internal.util.OperatingSystem;
+
 import java.util.Locale;
 
 /**
- * Supported platforms
+ * Supported OperatingSystem and Architecture.
  */
 public record Platform(OperatingSystem os, Architecture arch) {
 
-    public enum OperatingSystem {
-        WINDOWS,
-        LINUX,
-        MACOS,
-        AIX,
-        UNKNOWN;
-    }
-
-    public enum Architecture {
-        X86,
-        x64,
-        ARM,
-        AARCH64,
-        UNKNOWN;
-    }
-
-    public static final Platform UNKNOWN = new Platform(OperatingSystem.UNKNOWN, Architecture.UNKNOWN);
-
     /*
      * Returns the {@code Platform} based on the platformString of the form <operating system>-<arch>.
+     * @throws IllegalArgumentException if the delimiter is missing or either OS or
+     * architecture is not known
      */
     public static Platform parsePlatform(String platformString) {
         String osName;
         String archName;
         int index = platformString.indexOf("-");
         if (index < 0) {
-            osName = platformString;
-            archName = "UNKNOWN";
-        } else {
-            osName = platformString.substring(0, index);
-            archName = platformString.substring(index + 1);
+            throw new IllegalArgumentException("platformString missing delimiter: " + platformString);
         }
-        OperatingSystem os;
-        try {
-            os = OperatingSystem.valueOf(osName.toUpperCase(Locale.ENGLISH));
-        } catch (IllegalArgumentException e) {
-            os = OperatingSystem.UNKNOWN;
-        }
-        Architecture arch = toArch(archName);
+        osName = platformString.substring(0, index);
+        OperatingSystem os = OperatingSystem.valueOf(osName.toUpperCase(Locale.ROOT));
+
+        archName = platformString.substring(index + 1);
+        // Unalias Jmod architecture "amd64" to "x86_64"
+        archName = archName.replace("amd64", "X64");
+        Architecture arch = Architecture.valueOf(archName.toUpperCase(Locale.ROOT));
+
         return new Platform(os, arch);
     }
 
     /**
      * @return true is it's a 64-bit platform
      */
     public boolean is64Bit() {
-        return (arch() == Platform.Architecture.x64 ||
-                arch() == Platform.Architecture.AARCH64);
+        return (Architecture.is64bit());
     }
 
     /**
      * Returns the runtime {@code Platform}.
      */
     public static Platform runtime() {
-        return new Platform(runtimeOS(), runtimeArch());
+        return new Platform(OperatingSystem.current(), Architecture.current());
```

Comment:
Should we create a single `private static final Platform`, in this class, to represent the current platform, and keep returning it instead of creating a new instance on each call?